### ISSUE TITLE
parseUrl and makeUrlAbsolute incorrectly parse and generate URLs with the Windows 'res:' protocol.

### ISSUE THREAD

**Issue by bbesseling** [jquery/jquery-mobile#8355]
In a Windows app where we want to use jqm as the user interface in an IWebBrowser2,
when the document base is the Windows app resource URL
'res://device:\\directory\\file.exe/resource.html',
the function jQuery.mobile.path.parseUrl incorrectly parses the 'device' part (which could be e.g. 'C' or 'x-wmapp1')
into the authority, host and hostname fields.
This causes the function jQuery.mobile.path.makeUrlAbsolute to convert '#two' into
'res://device/:\\directory\\file.exe/resource.html#two'
with a spurious '/' after device.
This causes all non data-ajax="false" navigation to fail
The 'res://' protocol should also be treated like the 'file:///' protocol.
[respathissue.zip](https://github.com/jquery/jquery-mobile/files/84033/respathissue.zip)

**Comment by apsdehal** [comment]
JQM is not supposed to work with file and res protocols.

**Comment by arschmitz** [comment]
we do support native file protocols for apps like this

**Comment by arschmitz** [comment]
I think we fixed this already though might just be for 1.5 @gabrielschulhof do you recall?

**Comment by gabrielschulhof** [comment]
Yeah, there was an issue with Cordova on Windows, but I don't remember if it made it into 1.4.x or not.

**Comment by gabrielschulhof** [comment]
https://github.com/jquery/jquery-mobile/issues/6574

**Comment by apsdehal** [comment]
I can reproduce the issue still with latest 1.5-dev, I don't think it is fixed yet.

**Comment by gabrielschulhof** [comment]
The original problem I think was that the res protocol has no absolute path. That is, it looks like this:
```
res:path/to/file.html
```
And so it's hard to correctly identify the path, because ```/``` is assumed to follow the protocol to indicate an absolute path.

**Comment by gabrielschulhof** [comment]
Is it possible that there's a regression in Cordova?

**Comment by gabrielschulhof** [comment]
@apsdehal can you please detail the steps you used to reproduce it with 1.5-dev?

**Comment by apsdehal** [comment]
@gabrielschulhof I reproduce it with URI given in the issue description, though I am not sure if this URI is correct. Can you confirm this? 
`$.mobile.path.makeUrlAbsolute("#two", "res://device:\\directory\\file.exe/resource.html")`

**Comment by bbesseling** [comment]
yes, that is the URL from the original problem submission.
